nautobot/ui/src/components/Navbar.js

```diff
@@ -41,6 +45,35 @@ export function Navbar() {
     const location = useLocation();
     const { data: readyRoutes } = useGetNewUIReadyRoutesQuery();
 
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const changeHandler = (event) => {
+        let filters = [];
+        if (event.target.value !== "") {
+            filters = [["q", event.target.value]];
+        }
```

Comment:
Let's talk code style a bit:

1. Unless really required, most of the time it's better to use `event.currentTarget` rather than `target`. `currentTarget` references the node which the event was hooked up to and, thanks to that, it is guaranteed to be stable. `target`, on the other hand, holds a reference to a node which the event was fired on, which resolves during the application runtime, and makes it unstable. Even though it doesn't make much of a difference in `onChange` handlers, at the very least it guarantees type safety (that is, if TypeScript was used).

2. JavaScript coerces `""` to `false`, so the conditional statement can be simplified to `if (event.currentTarget.value)`.

3. Opinions are divided on this one, but I'd start with a known fact that functional programming doesn't allow variable reassignments or mutations (immutability). While JavaScript is not fully complaint with functional paradigm, functional is still a preferred way of writing JS code for the vast majority of engineers, simply because the lesser mutations and side effects there are in the source code, the easier it is to reason about. So, to make this statement look like functional programming, one could use a `const` and a ternary operator:
   ```
   const filters = event.currentTarget.value ? [["q", event.currentTarget.value]] : [];
   ```

These are just code style comments, the behavior remains unchanged. I do not insist on anything, because it's ultimately up to the team to agree on the code style to keep the source code cohesive, but this is just how a frontend engineer would've wrote it.

Reply:
@norbert-mieczkowski-codilime Can we talk about this tomorrow morning? I ran into an issue with addressing these lint issues all the way down.